src/props/createVaultSetupToken.js

```diff
@@ -37,19 +40,69 @@ export const getCreateVaultSetupToken = ({
   paymentSource: PaymentSource,
 |}): CreateVaultSetupToken => {
   const data = buildXCreateVaultSetupTokenData({ paymentSource });
-
+  const startTime = Date.now();
   return memoize(() => {
     if (!createVaultSetupToken) {
       throw new Error(`createVaultSetupToken undefined`);
     }
 
-    return createVaultSetupToken(data).then((vaultSetupToken) => {
-      if (!vaultSetupToken || typeof vaultSetupToken !== "string") {
-        throw new Error(
-          `Expected a vault setup token to be returned from createVaultSetupToken`
-        );
-      }
-      return vaultSetupToken;
-    });
+    return createVaultSetupToken(data)
+      .then((vaultSetupToken) => {
+        if (!vaultSetupToken || typeof vaultSetupToken !== "string") {
+          const errString = "Expected a vault setup token to be returned from createVaultSetupToken"
+          sendCountMetric({
+            name: "pp.app.paypal_sdk.buttons.create_vault_setup_token.count",
+            event: "error",
+            dimensions: {
+              errorName: 'no_setup_token',
+            }
+          })
+          getLogger()
+            .error('create_vault_setup_token', { err: errString })
+            .track({
+              [FPTI_KEY.STATE]:      FPTI_STATE.BUTTON,
+              [FPTI_KEY.ERROR_CODE]: 'smart_buttons_create_vault_setup_token',
+              [FPTI_KEY.ERROR_DESC]: errString
+            })
+          throw new Error(errString);
+        }
+        sendCountMetric({
+          name: "pp.app.paypal_sdk.buttons.create_vault_setup_token.count",
+          event: "success",
+        });
+        getLogger()
+          .addTrackingBuilder(() => {
+            return {
+              [FPTI_KEY.STATE]: FPTI_STATE.BUTTON,
+              [FPTI_KEY.TRANSITION]: FPTI_TRANSITION.RECEIVE_VAULT_SETUP_TOKEN,
+              [FPTI_KEY.EVENT_NAME]: FPTI_TRANSITION.RECEIVE_VAULT_SETUP_TOKEN,
+              [FPTI_KEY.CONTEXT_TYPE]: FPTI_CONTEXT_TYPE.VAULT_SETUP_TOKEN,
+              [FPTI_KEY.CONTEXT_ID]: vaultSetupToken,
+              [FPTI_CONTEXT_TYPE.VAULT_SETUP_TOKEN]: vaultSetupToken,
+              [FPTI_KEY.RESPONSE_DURATION]: (Date.now() - startTime).toString(),
+              client_time: getClientsideTimestamp(),
+            };
+          }).flush()
```

Comment:
Since this is a success, we can rely on a flush further in the flow
```suggestion
          })
```